### ISSUE TITLE
update packagist with recent changes

### ISSUE THREAD

**Issue by chadkouse** [polyfractal/sherlock#7]
I'm using composer to manage dependencies and so before I can push this code to my application cluster I will need the changes that were merged to master to show up in packagist.  I've never actually done that and imagine I'm not even able to, so I'm asking for your help with that.

**Comment by chadkouse** [comment]
although I suppose I could roll my own package.. I'd rather not :)

**Comment by polyfractal** [comment]
Tagged as v0.1.3, pushed to Github and Packagist.  Should automatically update to the latest when you run composer.  Let me know if you have any problems.

Thanks for being patient with me!

**Comment by chadkouse** [comment]
not a problem happy to see someone taking up this cause :)

**Comment by chadkouse** [comment]
I could use a packagist update since the logging updates were merged from #12

**Comment by polyfractal** [comment]
I was playing with Composer over the weekend and discovered you can do this in your composer.json:

``` json
{
    "require": {
        "sherlock/sherlock": "dev-master"
    }
}
```

Which will automatically track the `master` branch, updating whenever there is a change.  Since I only use master to track stable dev releases (even if they aren't tagged), that should give you the latest stable version.

Let me know if that works for you.  I'd like to group five to ten features/bugs before tagging a release, just so I'm not hitting crazy high numbers (0.1.205, etc).

But that's just cosmetic, so if it doesn't work I'll tag a release ASAP :)

**Comment by chadkouse** [comment]
Will do. It's probably not a best practice just because it basically subverts the composer.lock file - but for a project like yours that is undergoing rapid changes and bug fixes it probably makes sense for a little while.   

## 

Chad Kouse

**Comment by polyfractal** [comment]
Hmm...you still need to run `composer update` to fetch the most recent master, but I see why that's still not a good practice.

Do you think I should just tag each notable bug fix?  I guess cosmetics really don't matter, who cares if the numbers get high.  I'm new to this whole tag release schedule thing, any thoughts?

**Comment by chadkouse** [comment]
I was more talking about machines where we are installing the composer dependencies for the first time - it wouldn't be guaranteed to match the version on my dev machine (if I was a few commits behind for example)

Honestly I really don't know what the best approach is. I think bugs should be hotfixed and that probably means a new tag but I agree it could look weird and its probably annoying to make the packagist updates multiple times per day (potentially) but once things settle down it will be smoother.   

## 

Chad Kouse

**Comment by polyfractal** [comment]
Ah, I see what you are saying.  That that does pose a problem.  I hopped on the Composer IRC channel and the guys there had a few suggestions/thoughts:
- Main point is that I should use [Semantic Versioning](http://semver.org/), which ensures that all Minor/Patch versions are backwards compatible, and only Major versions break compatibility.
- Second point is that the 0.\* branch is basically exempt from semver because it doesn't guarantee any backwards compatibility.  Quote from one of the guys:  _"yeah, 0._\* _is reserved for "everything may break at any time" - it's sort of equivalent to using dev-master until you get to a release-able point"_

I think my gameplan is to start doing semver after the core feature-set is laid out, bump the Minor to 0.2 and then start following the minor/patch rules pretty religiously.

In the meantime, they offered some suggestions:
- Tag and release every bugfix/patch/release as a Patch version, only acceptable while in 0.*. 
  
  Its just a lot of overhead on my end, since I use a Git Flow model (develop -> release -> merge back into develop -> merge into master -> tag and push).  So doing that for each bugfix gets tedious.  Packagist updates automatically, so that isn't actually a problem.
- Change the requirement to dev-master
- Use a branch alias, which basically says "treat dev-master as the 0.1.\* branch".  You would then require 0.1.*@dev, which grabs the latest bugfixes from dev-master.  This is almost identical to requiring dev-master itself, except that when the Minor is bumped to 0.2.0 the branch alias changes as well, so it effectively keeps your boxes from upgrading to 0.2.0 without your explicit permission.

**Tl;dr:** I think I'm going to just keep tagging releases and deal with the hassle.  Hopefully I'll hit a core featureset soon and then it won't be as much of a problem.  I'll go add and push the latest tag now =)

**Comment by polyfractal** [comment]
v0.1.5 tagged and pushed =)

**Comment by chadkouse** [comment]
Thanks!  

-- chad
